scripts/get-aidrop-sources.ts

```diff
@@ -0,0 +1,222 @@
+import { ethers, upgrades } from "hardhat";
+import { BigNumber, Contract, ContractFactory } from "ethers";
+import Axios, { AxiosInstance } from "axios";
+import fs from "fs";
+import path from "path";
+const cliProgress = require('cli-progress');
+const _colors = require('colors');
+
+const alchemyUrl = 'https://eth-mainnet.alchemyapi.io/v2/6Kv_bJKux1FXeX5h7ga3GaxICthIx9UG';
+const etherscanUrl = 'https://api.etherscan.io/api';
+
+const tokenAddress = "0x0C63cAE5fcC2Ca3dDE60a35e50362220651eBEc8";
+const UNISWAP_stXEM = '0xab2f9b13b4aca6884fdbb9e4a1f767c82c53688f'
+const poolAddresses: string[] = [UNISWAP_stXEM];
+const blockNumber = undefined;
+
+async function getTokenBalance(
+  data: string,
+  contractAddress: string,
+  blockNumber?: number
+): Promise<BigNumber> {
+  const alchemyApi = Axios.create({
+    baseURL: alchemyUrl,
+    headers: {},
+    timeout: 120000,
+  });
+
+  const params: any = [{
+    data,
+    "to": contractAddress
+  }]
+
+  if (blockNumber) {
+    params.push('0x' + blockNumber.toString(16))
+  }
+
+  const result = await alchemyApi.post('', {
+    "id": 1,
+    "jsonrpc": "2.0",
+    "method": "eth_call",
+    params,
+  })
+
+  if (!result || !result.data || result.data.result === '0x') {
+    return ethers.BigNumber.from(0);
+  }
+
+  return result.data.result;
+}
+
+
+async function getTokenLogs(address: string): Promise<any[]> {
+  const ehterscanApi = Axios.create({
+    baseURL: etherscanUrl,
+    headers: {},
+    params: {
+      apiKey: process.env.ETHERSCAN_API_KEY,
+      fromBlock: 0,
+      toBlock: 'latest',
+      module: 'logs',
+      action: 'getLogs',
+      address
+    },
+    timeout: 120000,
+  });
+
+  const result = await ehterscanApi.get('');
+
+  if (!result || !result.data) {
+    return []
+  }
+
+  return result.data.result;
+}
+
+function getAddressIndexes(events: any[], eventName: string): number[] {
+  const eventData = events.filter((event: any) => eventName === event.name);
+
+  if (!eventData || eventData.length === 0) {
+    return []
+  }
+
+  return eventData[0].addressIndexes;
+}
+
+function getEventsFromAbi(abi: any[]) {
+  const events = abi.filter((obj: any) => obj.type ? obj.type === "event" : false).map((event: any) => {
+    const addressIndexes: number[] = [];
+    event.inputs.forEach((input: { name: string, type: string }, i: number) => {
+      if (input.type === 'address') {
+        addressIndexes.push(i);
+      }
+    })
+    return { ...event, addressIndexes }
+  })
+
+  return events;
+}
+
+
+
+async function main(): Promise<void> {
+  // Hardhat always runs the compile task when running scripts through it.
+  // If this runs in a standalone fashion you may want to call compile manually
+  // to make sure everything is compiled
+  // await run("compile");
+
+  const provider = new ethers.providers.AlchemyProvider('homestead', process.env.ALCHEMY_API_KEY);
+  let abi = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'stakedTokenABI.json')).toString());
```

Comment:
You can import files diectly instead of reading them as files

https://stackoverflow.com/questions/49996456/importing-json-file-in-typescript/49997144